code/game/objects/items/weapons/RMS_vr.dm

```diff
@@ -0,0 +1,231 @@
+					if(1)
+						product = pick(50; new /obj/item/weapon/cell/secborg/empty,
+										25; new /obj/item/weapon/cell/high/empty,
+										20; new /obj/item/weapon/cell/super/empty,
+										5; new /obj/item/device/fbp_backup_cell)
+					if(0)
+						product = pick(50; new /obj/item/weapon/cell/secborg/empty,
+										30; new /obj/item/weapon/cell/high/empty,
+										15; new /obj/item/weapon/cell/super/empty,
+										5; new /obj/item/device/fbp_backup_cell)
+				if(istype(product,/obj/item/weapon/cell/infinite))
+					consume_resources(max_charge)
+					spark_system.start()
+				else
+					consume_resources(charge_cost)
+			if(4)
+				if(!emagged)
+					product = pick(30;new /obj/item/stack/material/steel,
+									20;new /obj/item/stack/material/glass,
+									10;new /obj/item/stack/cable_coil/random,
+									10;new /obj/item/stack/material/plastic,
+									10;new /obj/item/stack/material/wood,
+									10;new /obj/item/stack/material/wood/sif,
+									10;new /obj/item/stack/material/log,
+									10;new /obj/item/stack/material/log/sif,
+									10;new /obj/item/stack/material/cloth,
+									10;new /obj/item/stack/material/cardboard,
+									10;new /obj/item/trash/rkibble)
+					consume_resources(charge_cost_r)
+					spark_system.start()
+				else
+					product = pick(10;new /obj/item/stack/cable_coil/random,
+									10;new /obj/item/stack/material/plastic,
+									10;new /obj/item/stack/material/wood,
+									10;new /obj/item/stack/material/wood/sif,
+									10;new /obj/item/stack/material/log,
+									10;new /obj/item/stack/material/log/sif,
+									10;new /obj/item/stack/material/cloth,
+									10;new /obj/item/stack/material/cardboard,
+									10;new /obj/item/trash/rkibble,
+									10;new /obj/item/weapon/reagent_containers/spray/waterflower,
+									10;new /obj/item/weapon/bikehorn,
+									10;new /obj/item/weapon/storage/backpack/clown,
+									10;new /obj/item/clothing/under/rank/clown,
+									10;new /obj/item/clothing/shoes/clown_shoes,
+									10;new /obj/item/clothing/mask/gas/clown_hat,
+									1;new /mob/living/simple_mob/vore/catgirl)
+					consume_resources(charge_cost)
+					spark_system.start()
+	product.loc = get_turf(A)
+
+//Start of attack functions
+
+/obj/item/weapon/rms/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+	if(istype(target, /obj/item/weapon/cell)) //Check for a battery on-click
+		drain_battery(user, target)
+		return
+	if(istype(target, /turf/simulated)) // Check for a proper area on-click to spawn items
+		use_rms(target, user)
+		return
+	else
+		to_chat(user, "<span class='notice'>Invalid target for the device.</span>")
+		return
+
+/obj/item/weapon/rms/attack_self(mob/user)
+	if(mode_index >= modes.len)
+		mode_index = 1
+	else
+		mode_index++
+
+	to_chat(user, span("notice", "Changed mode to '[modes[mode_index]]'."))
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+	return ..()
```

Comment:
Might want a radial menu like the RCD has.

Reply:
I will look into this.